src/model_repository_manager.h

```diff
@@ -350,10 +499,27 @@ class ModelRepositoryManager {
   Status PollAndUpdateInternal(bool* all_models_polled);
 
   /// The internal function that load or unload a set of models.
+  /// If 'no_parallel_conflict' is provided and there is a conflict, then the
+  /// function will block until the conflict is resolved and set the correct
+  /// value into the variable, and return with success status for retrying.
+  /// If 'no_parallel_conflict' is not provided and there is a conflict, then
+  /// the function will return immediately with an error status.
   Status LoadUnloadModels(
       const std::unordered_map<
           std::string, std::vector<const InferenceParameter*>>& models,
       const ActionType type, const bool unload_dependents,
+      bool* all_models_polled, bool* no_parallel_conflict = nullptr);
```

Comment:
For two reasons:
1. When the `LoadUnloadModels()` is called by `ModelRepositoryManager::Create()`, parallel conflict cannot exist since only the create function is manipulating the models, so `Create()` does not have to provide the parameter and can safely leave it at `nullptr`.
2. If we later decide to add the functionality of returning an error immediately back to the user when there is a load/unload conflict, then the caller can set the `no_parallel_conflict` to `nullptr`. The function will return an error `Status` immediately when a conflict is seen.

Reply:
1. But there is also no harm for `Create()` to provide `no_parallel_conflict` variable, right? It can do nothing with it, or actually sanity check on indeed nothing needs to be waited on. In the current implementation, your `LoadUnloadModels()` is slightly more complicated because the implementation needs to be aware that `no_parallel_conflict` may not be provided and need to add conditionals for the two scenarios.
2. Then isn't it the same for the caller to check if `*no_parallel_conflict == false` and decide the follow-up action (retry / abort)?

Reply:
1. Yes.
2. They are the same, except if the caller provides `no_parallel_conflict`, the `LoadUnloadModels()` will block until it is notified to retry (from the CV returned from `LockNodes()`) before returning. If `no_parallel_conflict == nullptr`, then the `LoadUnloadModels()` returns immediately (with an error `Status`) without waiting.